src/braket/analog_hamiltonian_simulator/rydberg/scipy_solver.py

```diff
@@ -105,27 +108,28 @@ def f(index_time: int, y: np.ndarray) -> scipy.sparse.csr_matrix:
         max_step=max_step,
         min_step=min_step,
     )
+    integrator.set_initial_value(state, 0)
 
-    if progress_bar:  # print a lightweight progress bar
+    if progress_bar:
         start_time = time.time()
+        update_interval = max(1, num_times // 100)
 
-    for index_time in range(len(simulation_times) - 1):
-        if progress_bar:  # print a lightweight progress bar
-            _print_progress_bar(len(simulation_times), index_time, start_time)
-
+    # Main integration loop
+    for index_time in range(num_times - 1):
         if not integrator.successful():
             raise Exception(
                 "ODE integration error: Try to increase "
                 "the allowed number of substeps by increasing "
                 "the parameter `nsteps`."
             )
 
-        integrator.set_initial_value(state, index_time)
         integrator.integrate(index_time + 1)
 
-        # get the current state, and normalize it
         state = integrator.y
-        state /= np.linalg.norm(state)  # normalize the state
-        states.append(state)
+        state /= np.linalg.norm(state)
+        states[index_time + 1] = state
+
+        if progress_bar and (index_time % update_interval == 0 or index_time == num_times - 2):
+            _print_progress_bar(num_times, index_time, start_time)
 
-    return states
+    return states
```

Comment:
nit: Add empty line at end of file
```suggestion
    return states

```